pydash/pydash_app/dashboard/aggregator/statistics.py

```diff
@@ -136,3 +134,104 @@ def append(self, endpoint_call, dependencies):
 
     def rendered_value(self):
         return date_dict({k: len(v) for k, v in self.value.items()})
+
+
+class FastestExecutionTime(Statistic):
+    dependencies = []
+
+    def empty(self):
+        return -1
+
+    def field_name(self):
+        return 'fastest_measured_execution_time'
+
+    def append(self, endpoint_call, dependencies):
+        if self.value < 0 or self.value > endpoint_call.execution_time:
+            self.value = endpoint_call.execution_time
+
+    def rendered_value(self):
+        return self.value
+
+
+class SlowestExecutionTime(Statistic):
+    dependencies = []
+
+    def empty(self):
+        return -1
+
+    def field_name(self):
+        return 'slowest_measured_execution_time'
+
+    def append(self, endpoint_call, dependencies):
+        if self.value < endpoint_call.execution_time:
+            self.value = endpoint_call.execution_time
+
+    def rendered_value(self):
+        return self.value
+
+
+class TDigestStatistic(Statistic):
+    dependencies = []
+
+    def __init__(self):
+        super().__init__()
+        self.value = self.empty()
+        self._digest = tdigest.TDigest()
+
+    def empty(self):
```

Comment:
This seems odd to me. Why is it `-1` here?

Reply:
In order for all of these execution time related to correctly return an error value to show that there is no data yet, I have chosen to define `empty()` as -1, as 0 might be seen as an actual (albeit [almost] impossible) value.

Reply:
As discussed: a normal new `TDigest()` would be fine here, especially if combined with the 'presentable' tactic we talk about below.

Reply:
And things like what you did are of course allowed, but you need to document them if you do, because currently it's just a magic number :grin: